module/actor/sheets/mixins/healMixin.js

```diff
@@ -0,0 +1,134 @@
+        };
+
+        await new DialogV2({
+            window: { title: game.i18n.localize('WITCHER.Heal.dialogTitle') },
+            content: await renderTemplate('systems/TheWitcherTRPG/templates/dialog/heal/heal-rest.hbs', dialogData),
+            modal: false,
+            buttons: [
+                {
+                    action: 'heal',
+                    label: game.i18n.localize('WITCHER.Heal.button'),
+                    callback: async () => {
+                        const isResting = document.querySelector('#resting').checked;
+                        const isSterilized = document.querySelector('#sterilized').checked;
+
+                        await this.recoverActor(
+                            isResting,
+                            isSterilized,
+                            dialogData,
+                            hp.value,
+                            hp.max,
+                            sta.unmodifiedMax
+                        )
+                    }
+                },
+                {
+                    action: 'cancel',
+                    label: game.i18n.localize('WITCHER.Button.Cancel'),
+                    callback: () => {}
+                }
+            ]
+        }).render({ force: true });
+
+        this.restDialogListener(document, totalRec, rec, dialogData, actualWoundList)
+    },
+
+    async updateHealAmount(totalRec, rec, dialogData, actualWoundList) {
+      const isResting = document.querySelector('#resting').checked;
+      const isSterilized = document.querySelector('#sterilized').checked;
+      const isHealingHand = document.querySelector('#healing-hand').checked;
+      const isHealingTent = document.querySelector('#healing-tent').checked;
+
+      totalRec = Math.floor(rec.max / 2);
+
+      if (isResting) {
+          totalRec = rec.max;
+          dialogData.isResting = true;
+      }
+
+      if (isSterilized) {
+          totalRec += 2;
+          dialogData.isSterilized = true;
+
+          if (actualWoundList.some(wound => !wound.sterilized)) {
+            dialogData.daysHealed =+ 3;
+          }
+      }
+
+      if (isHealingHand) totalRec += 3;
+
+      if (isHealingTent) totalRec += 2;
+
+      dialogData.totalRec = totalRec;
+      document.querySelector('#extra-info').textContent =
+          `${game.i18n.localize('WITCHER.Heal.totalRecover')} + ${totalRec}`;
+
+      dialogData.sterilizedClass = isSterilized ? '' : 'invisible';
+      document.querySelector('#sterilized-info').className = dialogData.sterilizedClass;
+  },
+
+    async recoverActor(isResting, isSterilized, dialogData, curHealth, maxHealth, unmodifiedMaxSta) {
+        await this.actor.update({
+          'system.derivedStats.hp.value': Math.min(curHealth + dialogData.totalRec, maxHealth),
+          'system.derivedStats.sta.value': unmodifiedMaxSta
+        });
+
+        const critList = Object.values(this.actor.system.critWounds).map(details => details);

```

Comment:
this does basically nothing. Why dont you do const critList = this.actor.system.critWounds?

I see that this was a fault in the original code. Can you improve this? Thanks